pkg/api/metrics.go

```diff
@@ -61,9 +67,26 @@ func (hs *HTTPServer) QueryMetricsV2(c *m.ReqContext, reqDto dtos.MetricRequest)
 		})
 	}
 
-	resp, err := tsdb.HandleRequest(c.Req.Context(), ds, request)
+	if !expr {
```

Comment:
Feels like this function QueryMetricsV2 and QueryMetrics could need some refactoring, there is some duplication between the two and this new function is a bit long and messy, for example the duplicate !expr conditions can maybe be reduced to one if building the queries loop was extracted to a function.

Reply:
Reduced it some. I'm not sure about sharing code between the two endpoints since I think there might be some differences in Q2? Currently every query object with a request has datasource info, but they will all be executed based on the first datasource in the list of queries. Which is confusing. I guess it is fine because generally as things requests are grouped by datasource anyways (with the exception of gel). 

So in short I am still not clear on how we want the V2 endpoint to work yet.

Reply:
yea, that is fine for now 👍